app/api/users.rb

```diff
@@ -115,6 +115,10 @@ class Users < Grape::API
           end
           action = new_role.id > old_role.id ? :promote_user : :demote_user
 
+          unless user_parameters[:email] =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
```

Comment:
Can we move this into the User model?

You can add a custom validation there which will cover any modification to the field - and raise exceptions on error.